sql-plugin/src/main/scala/com/nvidia/spark/rapids/lore/GpuLore.scala

```diff
@@ -197,6 +202,18 @@ object GpuLore {
           s"when ${RapidsConf.LORE_DUMP_IDS.key} is set."))
 
       val spark = SparkShimImpl.sessionFromPlan(sparkPlan)
+
+      Option(spark.sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)).foreach { executionId =>
+        loreOutputRootPathChecked.computeIfAbsent(executionId, _ => {
+          val path = new Path(loreOutputRootPath)
+          val fs = path.getFileSystem(spark.sparkContext.hadoopConfiguration)
+          if (fs.exists(path) && fs.listStatus(path).nonEmpty) {
```

Comment:
could avoid an RPC for exists by just calling listStatus and catching FileNotFoundException

Reply:
@liurenjie1024 I see the :+1: but no change in the code.

Reply:
I am OK with that, just was not sure whether you just forgot to change it. 
It will not be a bottleneck here. It is a good practice to avoid `exists` checks with APIs that throw FNF.